File: Seed/Java/src/main/java/Hayaa/ConfigSeed/AppSeed.java
package Hayaa.ConfigSeed;

import hayaa.basemodel.model.BaseData;
import hayaa.basemodel.model.FunctionOpenResult;
import hayaa.basemodel.model.FunctionResult;
import hayaa.common.JsonHelper;
import hayaa.common.StringUtil;

import static javafx.scene.input.KeyCode.T;

public class AppSeed {
    private static AppSeed _instance = new AppSeed();

    public static AppSeed Instance() {

        return _instance;

    }

    /// <summary>
    /// 初始化，并获取配置
    /// </summary>
    /// <returns></returns>
    public String InitConfig() {
        String result = "";
        try {
            //支持分布式配置系统则获取配置
            System.out.println("读取设置");
            ProgramDistributedConfig.Instance().initAppStartInit();
            AppLocalConfig seedConfig = ProgramDistributedConfig.Instance().GetSeedConfig();

        } catch (Exception ex) {
            return ex.getMessage();
        }
        return result;
    }

    public String Resetonfig() {
        String result = "";
        try {
            //支持分布式配置系统则获取配置
            ProgramDistributedConfig.Instance().ReAppConfig();
            AppLocalConfig seedConfig = ProgramDistributedConfig.Instance().GetSeedConfig();

        } catch (Exception ex) {
            return ex.getMessage();
        }
        return result;
    }

    public static void SetAppInstanceId(int appInstanceId) {
        ProgramDistributedConfig.Instance().SetAppInstanceId(appInstanceId);
    }

    public static FunctionOpenResult<AppLocalConfig> GetAppLocalConfig() {
        FunctionOpenResult<AppLocalConfig> r = new FunctionOpenResult<>();
        AppLocalConfig config = ProgramDistributedConfig.Instance().GetSeedConfig();
        if (config != null) {
            r.setData(config);
        }
        return r;
    }
    public static <T> FunctionOpenResult<T> GetConfig(int componentID, Class<T> valueType)
    {
        FunctionOpenResult<T> r = new FunctionOpenResult<T>();
        ComponentConfig config = ProgramDistributedConfig.Instance().GetComponentConfig(componentID);
        if (config != null)
        {
            if (!StringUtil.IsNullOrEmpty(config.getContent()))
            {
                r.setData(JsonHelper.DeserializeObject(config.getContent(),valueType));
            }
        }
        return r;
    }
    public static <T> FunctionOpenResult<T> GetAppRootConfig(Class<T> valueType)
    {
        FunctionOpenResult<T> r = new FunctionOpenResult<>();
        AppConfig config = ProgramDistributedConfig.Instance().getAppConfig();
        if (config != null)
        {
            if (!StringUtil.IsNullOrEmpty(config.getConfigContent()))
            {
                r.setData(JsonHelper.DeserializeObject(config.getConfigContent(),valueType));
            }
        }
        return r;
    }
}

File: v1/Seed/Netcore/Hayaa.ConfigSeed.Netcore/AppSeed.cs
using Hayaa.BaseModel;
using Hayaa.ConfigSeed.Standard.Component;
using Hayaa.ConfigSeed.Standard.Model;
using System;

namespace Hayaa.ConfigSeed.Standard
{
    /// <summary>
    /// 宿主引用插件类
    /// 调用InitProgram方法获取远程配置
    /// </summary>
    public class AppSeed
    {
        private static AppSeed _instance = new AppSeed();
       static AppSeed()
        {

        }
        public static AppSeed Instance
        {
            get
            {
                return _instance;
            }
        }
        /// <summary>
        /// 初始化，并获取配置
        /// </summary>
        /// <returns></returns>
        public string InitConfig()
        {
            string result = "";
            try
            {
                //支持分布式配置系统则获取配置
                ProgramDistributedConfig.Instance.InitAppConfig();
                var seedConfig = ProgramDistributedConfig.Instance.GetSeedConfig();

            }
            catch (Exception ex)
            {
                return ex.Message;
            }
            return result;
        }
        public String Resetonfig()
        {
            string result = "";
            try
            {
                //支持分布式配置系统则获取配置
                ProgramDistributedConfig.Instance.ReAppConfig();
                var seedConfig = ProgramDistributedConfig.Instance.GetSeedConfig();

            }
            catch (Exception ex)
            {
                return ex.Message;
            }
            return result;
        }
        /// <summary>
        /// 设置App实例运行Id
        /// </summary>
        /// <param name="appInstanceId"></param>
        public static void SetAppInstanceId(int appInstanceId)
        {
            ProgramDistributedConfig.Instance.SetAppInstanceId(appInstanceId);
        }
        /// <summary>
        /// 获取AppId
        /// </summary>
        /// <returns></returns>
        public static FunctionOpenResult<AppLocalConfig> GetAppLocalConfig() 
        {
            var r = new FunctionOpenResult<AppLocalConfig>();
            AppLocalConfig config = ProgramDistributedConfig.Instance.GetSeedConfig();
            if (config != null)
            {
               
                    r.Data = config;
               
            }
            return r;
        }
        /// <summary>
        /// 按照组件ID获取组件配置内容
        /// </summary>
        /// <typeparam name="T"></typeparam>
        /// <param name="componentID"></param>
        /// <returns></returns>
        public static FunctionResult<T> GetConfig<T>(int componentID) where T : BaseData, new()
        {
            var r = new FunctionResult<T>();
            var config = ProgramDistributedConfig.Instance.GetComponentConfig(componentID);
            if (config != null)
            {
                if (!string.IsNullOrEmpty(config.Content))
                {
                    r.Data =Newtonsoft.Json.JsonConvert.DeserializeObject<T>(config.Content);
                }
            }
            return r;
        }
        public static FunctionOpenResult<T> GetAppRootConfig<T>() 
        {
            var r = new FunctionOpenResult<T>();
            var config = ProgramDistributedConfig.Instance.GetAppConfig();
            if (config != null)
            {
                if (!string.IsNullOrEmpty(config.ConfigContent))
                {
                    r.Data = Newtonsoft.Json.JsonConvert.DeserializeObject<T>(config.ConfigContent);
                }
            }
            return r;
        }
    }
}
